query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -154,6 +186,65 @@ private void extractColumns(List<String> columns, Expression expression) {
     }
   }
 
+  private Duration parseDuration(String timeSeriesPeriod) {
+    String[] splitPeriodString = timeSeriesPeriod.split(":");
+    long amount = Long.parseLong(splitPeriodString[0]);
+    ChronoUnit unit = TimeUnit.valueOf(splitPeriodString[1]).toChronoUnit();
+    return Duration.of(amount, unit);
+  }
+
+  private Optional<Duration> findTimeRangeDuration(Filter filter, String timeFilterColumn) {
+
+    // time filter will always be present with AND operator
+    if (filter.getOperator() != Operator.AND) {
+      return Optional.empty();
+    }
+
+    Optional<Long> timeRangeStart =
+        filter.getChildFilterList().stream()
+            .filter(
+                childFilter ->
+                    this.isMatchingFilter(
+                        childFilter, timeFilterColumn, List.of(Operator.GE, Operator.GT)))
+            .map(matchingFilter -> matchingFilter.getRhs().getLiteral().getValue().getLong())
+            .findFirst();
+
+    Optional<Long> timeRangeEnd =
+        filter.getChildFilterList().stream()
+            .filter(
+                childFilter ->
+                    this.isMatchingFilter(
+                        childFilter, timeFilterColumn, List.of(Operator.LT, Operator.LE)))
+            .map(matchingFilter -> matchingFilter.getRhs().getLiteral().getValue().getLong())
+            .findFirst();
+
+    if (timeRangeStart.isPresent() && timeRangeEnd.isPresent()) {
+      return Optional.of(
```

Comment:
This was changed from the suggestion, and converts from millis -> seconds -> duration instead of directly from millis -> duration. Was there a reason for that? If so, please add a comment since it's more code and loses information.